### PR TITLE
Initial Multilabel support

### PR DESCRIPTION
**Pull request by Moritz-Wirth**
## Initial Multi-label support.

BirdSet Embedding based sample study [al-birdset-embeddings](https://github.com/Moritz-Wirth/al-birdset-embeddings).

- Introduce a new classifier that handles multi label classifiers
   - Handles basic interactions between active learning and multi label classification
   - How to handle multi output classifiers
-  2 new native multi-label pool strategies from literature
   - MMC
   - Label Cardinality Inconsistency
- 6 adjusted classes
   - CLUE
   - DropQuery
   - GreedySamplingX
   - RandomSampling
   - UncertaintySampling
   - SubSamplingWrapper
- 3 new tutorial notebooks
   - Toy example (07_...)
   - BirdSet example (08_...)
   - NLP example (09_...)

Additionally, some utility and checking methods are adjusted to allow for multi-label data, while keeping the previous functionality.

### CODE REVIEW

File: skactiveml/base.py [hunk 1]
```diff
@@ -297,6 +298,8 @@ def _validate_data(
             provided when reset was last True.
         **check_X_dict : kwargs
             Parameters passed to :func:`sklearn.utils.check_array`.
+        is_multilabel : bool, default=False
+            Weather provided data is in multi-label format or not.
```
Comment (tpham93) [inline skactiveml/base.py:302]:
```suggestion
            If true, provided `y` must be in multi-label format.
```

File: skactiveml/base.py [hunk 2]
```diff
@@ -322,12 +325,22 @@ def _validate_data(
         ) = super()._validate_data(
             X, y, candidates, batch_size, return_utilities, reset, check_X_dict
         )
-        y = column_or_1d(y, warn=True)
+
+        y = column_or_1d(y, warn=True) if not is_multilabel else check_array(y, ensure_2d=True, force_all_finite="allow-nan")
```
Comment (tpham93) [inline skactiveml/base.py:329]:
```suggestion
        if is_multilabel:
            y = check_array(y, ensure_2d=True, force_all_finite="allow-nan")
        else:
            y = column_or_1d(y, warn=True)
```

File: skactiveml/base.py [hunk 3]
```diff
@@ -322,12 +325,22 @@ def _validate_data(
...
+        # here check
+        if is_multilabel and not np.all(np.isin(y, [0, 1]) | np.isnan(y)): # possible 0,1-encoded check
```
Comment (tpham93) [inline skactiveml/base.py:331]: - Usage of is_unlabeled with self.missing_label
- Unlabeled instances must contain only missing_label for every class
- Raise Error in case of inconsistencies

File: skactiveml/base.py [hunk 4]
```diff
@@ -322,12 +325,22 @@ def _validate_data(
...
+            raise ValueError("All elements in y must be either 0 or 1 for multilabel classification.")
```
Comment (tpham93) [inline skactiveml/base.py:332]:
```suggestion
            raise ValueError("All elements in `y` must be either 0, 1 or `missing_label` for multilabel classification.")
```

**Review by tpham93 — CHANGES_REQUESTED** [review]

**Review by tpham93 — APPROVED** [review]
